From 6b7fb950cd91ff020514b70eeed744026c2debc3 Mon Sep 17 00:00:00 2001
From: Wang ShaoBo <bobo.shaobowang@huawei.com>
Date: Thu, 8 Dec 2022 21:45:11 +0800
Subject: [PATCH] arm64/mpam: decrease dom_num when domain goes offline
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 6b7fb950cd91ff020514b70eeed744026c2debc3
Modified-by-SEL: No


hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I61CPK
CVE: NA

--------------------------------

mpam_resctrl_setup_domain() increase dom_num when domain goes online. As
a result, mpam_resctrl_cpu_offline() should decrease dom_num when domain
goes offline.

Signed-off-by: Wang ShaoBo <bobo.shaobowang@huawei.com>
Signed-off-by: Jialin Zhang <zhangjialin11@huawei.com>
Reviewed-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm64/kernel/mpam/mpam_setup.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm64/kernel/mpam/mpam_setup.c b/arch/arm64/kernel/mpam/mpam_setup.c
index d30910e0cda2..6e71c99d19b0 100644
--- a/arch/arm64/kernel/mpam/mpam_setup.c
+++ b/arch/arm64/kernel/mpam/mpam_setup.c
@@ -172,6 +172,8 @@ int mpam_resctrl_cpu_offline(unsigned int cpu)
 		list_del(&d->list);
 		dom = container_of(d, struct mpam_resctrl_dom, resctrl_dom);
 		kfree(dom);
+
+		res->resctrl_res.dom_num--;
 	}
 
 	mpam_resctrl_clear_default_cpu(cpu);
-- 
2.33.0

